src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,78 @@
+import { FC } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todos: Todo[],
+  tempTodo: Omit<Todo, 'id'> | null,
+  deleteTodo: (todoId: number) => void,
+}
+
+export const TodoList: FC<Props> = (props) => {
+  const {
+    todos,
+    tempTodo,
+    deleteTodo,
+  } = props;
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => (
+        <div
+          data-cy="Todo"
+          className={cn('todo', { completed: todo.completed })}
+          key={todo.id}
+        >
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+            />
+          </label>
+
+          <span data-cy="TodoTitle" className="todo__title">
+            {todo.title}
+          </span>
+
+          <button
+            type="button"
+            className="todo__remove"
+            data-cy="TodoDelete"
+            onClick={() => deleteTodo(todo.id)}
+          >
+            ×
+          </button>
+
+          {/* overlay will cover the todo while it is being updated */}
+          <div data-cy="TodoLoader" className="modal overlay">
```

Comment:
Remove comment and show loader when user delete or toggle all todos, so user can understand that they should wait